### ISSUE TITLE
NVDA does not announce all details of a JTree: item selection issue (Java Access Bridge API)

### ISSUE THREAD

**Issue by anOolamr** [nvaccess/nvda#13908]
NVDA does not announce 
- the details of a JTree except a tree item is selected,
- selection or deselection of tree items,
- multiline-selection of tree items.

### Steps to reproduce:

1. If not available, download the SwingSet2 Demonstration Program from: https://docs.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html#swingset2

2. Start NVDA.

3. Launch SwingSet2.jar with an up-to-date Java JRE.
-> The SwingSet2 application window appears.

4. Navigate with arrow right key to the "JTree demo" radio button and press Spacebar key to select it.

5. Press Tab key twice to navigate with the keyboard focus to the JTree control.
-> NVDA announces "tree view", which is the role of this JTree control.
-> NVDA should announce the role, details regarding the root node and for example that no tree item is selected.

6. Press the command Ctrl + arrow down to set the keyboard focus on the root node. 
-> NVDA announces nothing.
-> NVDA should announce "Music, expanded, editable, 1 of 1, level 1".

7. Press Ctrl + Tab to move the keyboard focus backwards out of the JTree and then press Tab key to move the keyboard focus to the JTree with the focused root node.
-> NVDA announces "tree view".
-> NVDA should announce "tree view, Music, expanded, editable, 1 of 1, level 1", as the root node has the keyboard focus.

8. Press the command Ctrl + Spacebar to select the focused tree item.
-> NVDA announces nothing.
-> NVDA should announce "Music, selected, expanded, editable, 1 of 1, level 1".

9. Press Ctrl + Tab to move the keyboard focus backwards out of the JTree and then press Tab key to move the keyboard focus back to the JTree with the selected root node.
-> NVDA announces "tree view, Music, selected, expanded, editable, 1 of 1, level 1".

10. Press the command Ctrl + Spacebar to deselect the root node.
-> NVDA announces nothing.
-> NVDA should announce "Music, expanded, editable, 1 of 1, level 1".

11. Press the command Ctrl + Spacebar to select the root node and then the command Shift + Arrow down to select the child node "Classical".
-> NVDA announces the last selected tree item.
-> NVDA should announce the last selected tree item and that two items are selected.

12. Press Ctrl + Tab to move the keyboard focus backwards out of the JTree and then press Tab key to move the keyboard focus to the JTree with the selected tree items.
-> NVDA announces "tree view" and one of the selected items.
-> NVDA should additionally announce that two tree items are selected.

### Actual behavior:
When navigating to a JTree, NVDA does not announce tree details except at least one tree item is selected.
NVDA does not announce the tree item when selecting and deselecting it.
NVDA does not announce multiline-selection of tree items.

### Expected behavior:
When navigating to a JTree, NVDA announces also tree details when no tree item is selected.
NVDA announces the selection and deselection of tree items.
NVDA announces multiline-selection of tree items.

#### NVDA installed/portable/running from source:
installed

#### NVDA version:
Version: 2022.1

#### Windows version:
Version 10.0.19043 Build 19043

#### Name and version of other software in use when reproducing the issue:
java version "18.0.1" 2022-04-19
Java(TM) SE Runtime Environment (build 18.0.1+10-24)
Java HotSpot(TM) 64-Bit Server VM (build 18.0.1+10-24, mixed mode, sharing)

#### Other information about your system:
Java Access Bridge is enabled.

#### Does the issue still occur after restarting your computer?
Yws.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. Same behavior.

#### If NVDA add-ons are disabled, is your problem still occurring?
Not applicable.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Not applicable.

**Comment by seanbudd** [comment]
I believe this is fixed in 2022.2rc1 by #10993

Could you please test with 2022.2rc1 (or the latest alpha) before reporting these Java issues?

**Comment by anOolamr** [comment]
@seanbudd Thanks for your hint!
I have retested with version 2022.2rc1 (by running nvda_2022.2rc1.exe), downloaded from:
https://www.nvaccess.org/files/nvda/snapshots/
The described issues still exist.

**Comment by Adriani90** [comment]
This is still reproducible in NVDA 2024.4.1.
In fact, when pressing ctrl+space bar to unselect anitem after using shift+arrow keys and ctrl+up and down arrows to navigate through the selected items, NVDA does the oposite of the behavior in Windows explorer.
1. Unselected items are announced without any atribute, while in Windows explorer unselected items are announced as "not selected"
2. Selected items are announced as "selected" while in Windows Explorer selected items are not announced with the selected atribute.

I wonder whether the focus redirection is correct here. Java should manage the focus redirection on the treeview, so is the element really focused with the system focus when I land on the tree view after pressing tab key? If the system focus is redirected to a node outside of the actual object, then NVDA will not announce the selected item because it is not focused.

@anOolamr you wrote:
> NVDA should announce the role, details regarding the root node and for example that no tree item is selected.

The "no item selected" message needs to be exposed by Java Access bridge. Otherwise NVDA will not be able to calculate this itself I guess. The same is valid for the message of how many items are selected. This should be ideally calculated and exposed as a message by the API.

But maybe I am wrong. cc: @dmitrii-drobotov maybe you also have some suggestions her?
